nion/utils/ListModel.py

```diff
@@ -458,8 +465,12 @@ def __insert_item(self, item: typing.Any, sort_key: OptionalSortKeyCallable) ->
             for selection in self.__selections:
                 selection.insert_index(before_index)
 
-    def __remove_item(self, item: typing.Any) -> None:
-        item_index = self.__items.index(item)
+    def __remove_item(self, item: typing.Union[typing.Any, int]) -> None:
```

Comment:
Is there any reason this model would contain ints? If so this function would do the wrong thing if you were trying to remove a particular int from the list by assuming you were removing at a particular index.

An alternative would be to split this into 2 functions - `__remove_item`, and `__remove_item_at_index`

Looking at how this is currently used, `__remove_first` would be an alternative as well.